interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/TypeInfos.scala

```diff
@@ -23,54 +24,63 @@ object TypeInfos {
   sealed trait MethodInfo {
     def computeResultType(arguments: List[TypingResult]): ValidatedNel[ExpressionParseError, TypingResult]
 
-    def staticParameters: ParameterList
+    def signatures: NonEmptyList[MethodTypeInfo]
 
-    final def staticNoVarArgParameters: List[Parameter] = staticParameters.noVarArgs
-
-    final def staticVarArgParameter: Option[Parameter] = staticParameters.varArg
-
-    def staticResult: TypingResult
+    final def mainSignature: MethodTypeInfo = signatures.toList.maxBy(_.parametersToList.length)
 
     def name: String
 
     def description: Option[String]
 
-    final def varArgs: Boolean = staticVarArgParameter.isDefined
-
     final def serializable: SerializableMethodInfo =
-      SerializableMethodInfo(staticParameters.toList, staticResult, description, varArgs)
+      SerializableMethodInfo(mainSignature.parametersToList, mainSignature.result, description, mainSignature.varArg.isDefined)
+
+    protected def convertError(error: GenericFunctionTypingError, arguments: List[TypingResult]): ExpressionParseError =
+      SpelExpressionParseErrorConverter(this, arguments).convert(error)
   }
 
-  case class StaticMethodInfo(staticParameters: ParameterList,
-                              staticResult: TypingResult,
+  object StaticMethodInfo {
+    def apply(signatures: MethodTypeInfo, name: String, description: Option[String]): StaticMethodInfo =
+      StaticMethodInfo(NonEmptyList.one(signatures), name, description)
+  }
+
+  case class StaticMethodInfo(signatures: NonEmptyList[MethodTypeInfo],
```

Comment:
IMO would be better to replace make computing mainSignatures eager. I mean here pass one signature instead of Nel of signatures, remove mainSignature and explicitly named method like `StaticMethodInfo.flatten(signatures: MethodTypeInfo)` flatten/generalized/widened - I don't like "main" word in this context. Thanks to this approach this class will look like simple DTO - serialize/deserialize will create the same object.